iree/compiler/Dialect/VM/Analysis/ValueLiveness.cpp

```diff
@@ -47,7 +47,8 @@ LogicalResult ValueLiveness::annotateIR(IREE::VM::FuncOp funcOp) {
   // Block names are their order in the function.
   DenseMap<Block *, int> blockOrdinals;
   for (auto &block : funcOp.getBlocks()) {
-    blockOrdinals[&block] = blockOrdinals.size();
+    int ordinal = blockOrdinals.size();
+    blockOrdinals[std::addressof(block)] = ordinal;
```

Comment:
Why use `std::adressof`? Are we concerned about overloads of the `&` operator here??

Reply:
more readable for what we are doing (it's annoying that getBlocks returns references instead of pointers like all the other functions dealing with blocks, and taking addresses of references is kind of sketchy) - this ensures that we get compile errors if we were trying to take the address of a stack-local Block (vs the Block& from the iterator).